src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,159 @@
+#include "resultsdisassemblypage.h"
+#include "ui_resultsdisassemblypage.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QFile>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QString>
+#include <QListWidgetItem>
+#include <QProcess>
+#include <QDebug>
+#include <QDir>
+#include <QDirIterator>
+
+#include <KRecursiveFilterProxyModel>
+
+#include "parsers/perf/perfparser.h"
+#include "resultsutil.h"
+
+#include "models/costdelegate.h"
+#include "models/hashmodel.h"
+#include "models/topproxy.h"
+#include "models/treemodel.h"
+
+#include <QStandardItemModel>
+
+ResultsDisassemblyPage::ResultsDisassemblyPage(FilterAndZoomStack *filterStack, PerfParser *parser, QWidget *parent)
+        : QWidget(parent), ui(new Ui::ResultsDisassemblyPage) {
+    ui->setupUi(this);
+    ui->splitter->setStretchFactor(0, 1);
+    ui->splitter->setStretchFactor(1, 1);
+    ui->splitter->setStretchFactor(2, 8);
+}
+
+ResultsDisassemblyPage::~ResultsDisassemblyPage() = default;
+
+/**
+ *  Clear
+ */
+void ResultsDisassemblyPage::clear() {
+    if (ui->asmView->model() == nullptr) return;
+
+    int rowCount = ui->asmView->model()->rowCount();
+    if (rowCount > 0) {
+        ui->asmView->model()->removeRows(0, rowCount, QModelIndex());
+    }
+}
+
+/**
+ *  Produce and show disassembly with 'objdump'
+ */
+void ResultsDisassemblyPage::showDisassembly() {
+    showDisassemblyByAddressRange();
+}
+
+/**
+ *  Produce disassembler with 'objdump' by addresses range and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassemblyByAddressRange() {
+    // Show empty tab when selected symbol is not valid
+    if (m_curSymbol.symbol.isEmpty()) {
+        clear();
+        return;
+    }
+
+    // Call objdump with arguments: addresses range and binary file
+    QString processName =
+            m_objdump + QLatin1String(" -d --start-address=0x") + QString::number(m_curSymbol.relAddr, 16) +
+            QLatin1String(" --stop-address=0x") +
+            QString::number(m_curSymbol.relAddr + m_curSymbol.size, 16) +
+            QLatin1String(" ") + m_curSymbol.path;
+
+    showDisassembly(processName);
+}
+
+/**
+ *  Produce disassembler with 'objdump' and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassembly(QString processName) {
+    QTemporaryFile m_tmpFile;
+    QProcess asmProcess;
+
+    if (m_tmpFile.open()) {
+        asmProcess.start(processName);
+
+        if (!asmProcess.waitForStarted() || !asmProcess.waitForFinished()) {
+            return;
+        }
+        QTextStream stream(&m_tmpFile);
+        stream << asmProcess.readAllStandardOutput();
+        m_tmpFile.close();
+    }
+
+    if (m_tmpFile.open()) {
+        int row = 0;
+        QStandardItemModel *model = new QStandardItemModel();
```

Comment:
please don't use QStandardItemModel, implement a real one (in a separate file) that operates on a vector of structs that hold the actual data

Reply:
I enter DisassemblyModel in commit e68b7b7d0c5515c4b5e69f8058d1b31fa0c5ac2d. Please consider QStandardItemModel  as temporary.